### PR TITLE
PIM-8463: display an explicit error message when an uploaded file is too big on import profiles

### PR DESCRIPTION
**Pull request by rouliane**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by Doodoune** [comment]
Please add a Changelog :)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 1]
```diff
@@ -401,15 +402,24 @@ protected function launchAction(Request $request, string $code): Response
             throw new AccessDeniedHttpException();
         }
 
+        /** @var UploadedFile|null $file */
```
Comment (phaseinducer) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: please no

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 2]
```diff
@@ -638,4 +648,17 @@ protected function createAction(Request $request, string $type)
 
         return new JsonResponse($this->normalizeJobInstance($jobInstance));
     }
+
+    /**
+     * This is the only way we found to test that a file too big was uploaded.
+     * If the file size exceeds the server limit, a warning is thrown on the apache container logs
+     * and the request does not contain any information that a file was uploaded on the FPM side.
```
Comment (phaseinducer) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: I understand this explaination but not how it's linked to `$_SERVER['CONTENT_LENGTH']`.

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 3]
```diff
@@ -638,4 +648,17 @@ protected function createAction(Request $request, string $type)
...
+     * Note: it happens only when the upload exceeds 'post_max_size'
+     *
+     * @return bool
+     */
+    private function isFileUpload(): bool
+    {
+        return isset($_SERVER['CONTENT_LENGTH']) && $_SERVER['CONTENT_LENGTH'] > 0;
```
Comment (nidup) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: You may access server global variables from Symfony service .

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php (hunk 2 above)
Comment (jmleroux) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: I added some explanation

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 4]
```diff
@@ -402,14 +403,22 @@ protected function launchAction(Request $request, string $code): Response
         }
 
         $file = $request->files->get('file');
+        if (null === $file && $this->isFileUpload($request)) {
+            return new JsonResponse(['message' => 'pim_import_export.entity.import_profile.flash.upload.error_too_big'], 400);
```
Comment (BitOne) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: The error maybe totally something else than the size.

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php [hunk 5]
```diff
@@ -402,14 +403,22 @@ protected function launchAction(Request $request, string $code): Response
...
+        }
+
         if (null !== $file) {
+            if (UPLOAD_ERR_INI_SIZE === $file->getError()) {
```
Comment (BitOne) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Controller/InternalApi/JobInstanceController.php]: Other errors can happen too: https://www.php.net/manual/en/features.file-upload.errors.php

So either we should manage them one by one, or we must provide a generic error message, like: "Unable to upload the file. Maybe its size exceeds the configured upload max size on your PHP or HTTP server configuration?"

File: CHANGELOG-3.0.md [hunk 6]
```diff
@@ -10,6 +10,7 @@
 
 - PIM-8413: Fix modal of category selection in product and product model exports
 - PIM-8477: Fix rich text area link dialog
+- PIM-8463: display an explicit error message when an uploaded file is too big on import profiles
```
Comment (BitOne) [inline CHANGELOG-3.0.md]:
```suggestion
- PIM-8463: On import profiles, displays an explicit error message when file upload fails, for example when the file is too big
```

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.en_US.yml [hunk 7]
```diff
@@ -31,6 +31,8 @@ pim_import_export:
             flash:
                 create:
                     success: Import profile successfully created
+                upload:
+                    error: Unable to upload the file. Its size may exceed the configured upload max size on your PHP or HTTP server configuration?
```
Comment (BitOne) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                    error: Unable to upload the file. Its size may exceed the configured maximum upload size on your PHP or HTTP server configuration
```

**Pull request merged**